docs/logging.md

```diff
@@ -0,0 +1,165 @@
+---
+id: logging
+title: Logging
+---
+
+### Logging
+
+Almost every application nowadays does logging, whether it's logging of business events, history of some state of 
+application or even debug record sequences. We log to files, distributed systems, standard outputs of OSes, and we 
+got used to it so much it seems really simple to us. Many existing libraries provide many ways of logging from 
+your application's code, either automatic or manual.
+
+### What's wrong (and how to fix that)?
+
+First, logging is, in fact, a side effect. Whether we log anything to a standard output, or a file - it's an interaction 
+with the outside world ('outside' relative to our program). Thus, logging as a process can be represented as an effect and operations - as an algebra.  
+Second, logging itself seems easy, but stuff around it - not so much. MDC, contexts, structured logging - all these 
+things are very helpful on a day-to-day basis, allowing us to read and write our logs, making them clean, comprehensible 
+and useful for many purposes. But there are few simple ways to provide all this functionality without bleeding and 
+screaming from seeing over-engineered solutions.  
+Tofu offers an easy and understandable abstraction over logging (it's even named like this!) but keeping in mind that 
+logging is an effect that can be composed.  
+It looks very similar to interfaces we all are used to (from SLF4J and other logging libraries and facades), introducing some new concepts:
+```scala mdoc
+import tofu.logging.LoggedValue
+
+trait LoggingBase[F[_]] {
+    def trace(message: String, values: LoggedValue*): F[Unit]
+    def debug(message: String, values: LoggedValue*): F[Unit]
+    def info(message: String, values: LoggedValue*): F[Unit]
+    def warn(message: String, values: LoggedValue*): F[Unit]
+    def error(message: String, values: LoggedValue*): F[Unit]
+}
+```
+
+Basically, that's the main user API you can interact with throughout your code. 
+It resembles well-known solutions, keeping in mind that logging is an effect, though introducing the concept of `LoggedValue`.  
+
+
+
+#### Logging creation
+Creating an instance of `Logging` in Tofu is an effect itself. You can use helper class `Logs` that can produce values of
+type `I[Logging[F]]` (where `I` and `F` can be two different effects) and define a behaviour of your `Logging` instances, 
+for example:  
+```scala mdoc
+import tofu.logging._
+
+// for simplification, you can use whatever F you like
+type F[A] = cats.effect.IO[A]
+
+class MyService
+
+// defines simple logging behaviour without any notion of contexts
+val syncLogs = Logs.sync[F, F]
```

Comment:
Is it a 'heavy' operation?

Reply:
No it's not. But we advice users to share an instance of `Logs`, maybe keep a single one for a whole application.

Reply:
As for me it needs a clarification.